.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,32 @@
+**Note: for support questions, please use stackoverflow**.
+This repository's issues are reserved for feature requests and bug reports.
+
+* **Type of issue**
+  - [ ] bug report
+  - [ ] feature request
+  - [ ] other enhancement
+
+* **Development Phase**
```

Comment:
What does this mean for a pull request that has associated code? Are you trying to categorize these as "implementation sketch" vs. "production / merge-ready code"?

Reply:
Here's Danny's email comment on this (he's welcome to replace this with a real comment):

> I think a "request" will be generally in the form of a suggestions to improve a feature or a bug report where as a "proposal" would be a proposed solution to a "request" that needs more discussion. 
> 
> For example, if some one says "It would be great if we could add annotations, here are what I want to use them for", that would be a "request". And then if some one else says "here's a approach to an annotation system", that would be a "proposal". 
> 
> Also, some of the current pull requests in the Chisel repo such as PR#694 and PR#645 should be categorized as "proposal" rather than "implementation" because the high level approach is still under discussion. Pull requests such as PR#485 should be categorized as "implementation" because people agree on the approach and the PR essentially just needs a low level code review/test coverage to be merged in.
> 
> Hopefully good usage of this can get people to discuss their use cases and resolve their differences at the proposal stage and then one party can make a implementation that everyone will be happy with.